subprojects/android-test/ui-testing-core/src/main/kotlin/com/avito/android/test/page_object/KeyboardElement.kt

```diff
@@ -26,98 +27,65 @@ class KeyboardElement : PageObject() {
         }
     }
 
-    /**
-     * WARNING: we can't check keyboard status after rotating screen using programmatically way.
-     * For this case, we can't detect keyboard open status and always returns false for
-     * isDisplayed method and true for isNotDisplayed method.
-     *
-     * https://issuetracker.google.com/u/1/issues/68137674
-     */
     class CheckLibrary {
 
-        fun isDisplayed(activity: Activity) {
-            // FIXME(MBS-1301)
-            Thread.sleep(1000)
-            checkDisplayed(activity, true)
-        }
-
-        fun isNotDisplayed(activity: Activity) {
-            // FIXME(MBS-1301)
-            Thread.sleep(1000)
-            checkDisplayed(activity, false)
-        }
-
-        private fun checkDisplayed(activity: Activity, isOpen: Boolean) {
-            var threshold = 0
-            var activityEffectiveHeight = 0
+        /**
+         * For backward compatibility
+         */
+        fun isDisplayed(@Suppress("UNUSED_PARAMETER") activity: Activity?) = isDisplayed()
 
-            InstrumentationRegistry.getInstrumentation().runOnMainSync {
```

Comment:
I know that this is a more honest way, but I don't know how to do it more stable. The new one seems to be working.

Reply:
Did you evaluate flakiness of the new implementation?
If I'm not mistaken, `UiDevice` invocations were not so stable.